### PR TITLE
[doc] Mention when compiling happens in Heroku instructions

### PR DESCRIPTION
**Pull request by nruth**
based on https://github.com/thoughtbot/ember-cli-rails/issues/413

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,8 @@ You are ready to deploy:
 $ git push heroku master
 ```
 
+EmberCLI compilation happens at deploy-time, triggered by the `asset:precompilation` rake task.
```
Comment (seanpdoyle) [inline README.md]: Do you think this should be in a more prominent place?

Could you replace `EmberCLI` with `EmberCli`?

Isn't the Rake task `assets:precompile`?

Comment (nruth) [inline README.md]: I copy-pasted `EmberCLI` and `precompilation` from the Capistrano instructions to try to keep things consistent. Should I change them too?

Somewhere more prominent would be nice, especially if it's a general feature of the gem and not only something that happens on Heroku. It seems like a really nice architectural feature / benefit and worth telling people about.

Comment (seanpdoyle) [inline README.md]: the `EmberCLI` can stay for now, but the `precompilation` is incorrect and should be fixed elsewhere. Thanks!

> It seems like a really nice architectural feature / benefit and worth telling people about.
I agree! I'm open to suggestions on where to mention it.

**Comment by nruth** [comment]
How about restructuring the deploy section a little? It currently opens with CDN advice, which is sound advice, but maybe not the first thing the reader is looking for. 

I'd try promoting the "Deployment Strategies" section to introduce the Deployment section, add mention of the `rake assets:precompile` -> EmberCLI build solution, 
move the CDN advice to a subsection, and keep Heroku and Capistrano as they are.

**Pull request merged**